Prozess.txt

```diff
@@ -0,0 +1,240 @@
+
+	Standartausgaben:
+		Über Standartausgaben kann ein Programm Daten ausgeben.
+			Standartausgaben die Gerätedatei /dev/stdout, Deskriptor Nummer 1
+			Bsp. $find . -name "*html" > ausgaben.txt
+			Find wird in die datei ausgaben.txt umgeleitet.
+	
+	Standarfehlerausgabe:
+		Zweiter Ausgabestrom, der für Fehler- und Statusmeldungen gedacht ist.
+			Standartfehlerausgaben haben die Gerätedatei /dev/stderr, Deskriptor Num. 2
+
+	Pipe:
+		Eine Pipe / Pipeline ist ein Datenstrom zwischen zwei Prozessen.
+		Funktioniert nach dem first in first out Prinzip.
+		Das ergebnis eines Programms wird dabei als Eingabe eines anderen Programms verwendet.
+		In der shell wird dies durch den "|" realisiert. So kann man mehrere befehle hintereinander Ausführen.
+
+/dev/null:
+	Verhält sich wie eine leere Datei, mit dem Unterschied, dass eingehende Datenströme verworfen werden. 
+	Beim Lesezugriff wir ein einzelnes End-of-file Zeichen ausgegebn.
+	Verwendung:
+		-Verwerfung von nicht interessanten Datenströmen, z.B. asugaben vom Prozesor
+		
+Administratorrechte Prozess:
+	sudo chmod n+s (programmpfad) bewirkt dass das programm mit den rechten der Benutzergruppe des SuperUsers
+	ausgeführt wird. Generell steht sudo für SuperUser do, und führt den nachfolgenden befehl 
+	mit Administratorrechten aus(falls vorhanden).
+	
+.EXE Datei und ihre Struktur:
+	Bei einer .exe datei handelt es sich unter Windows-Basierten Systemen um eine Ausführbare datei.
+	Sie besteht aus:
+		-Ausführbarem Code, welcher in der .exe als .text vorliegt
+		-Datensektionen, liegt als .data vor
+		-Ressourcensektionen (.rsrc)
+		-Importdatensektion (.idata)
+		-Debuginfotmationssektion(.debug)
+
+Interprozesskommunikation:
+	Erfolgt über verschiedene Verfahren:o clarify
+		-Shared Memory:
+			-Mehrere Prozesse greifen auf einen gemeinsamen Datenspeicher zu, bsp. auf
+			 gemeinsame Bereiche des Arbeitsspeichers.
+		-Message Queue:
+			-Nachrichten von einem Prozess an eine Nachrichtenanlage geschickt, von
+			 wo ein anderer Prozess sie abholen kann.
+
+	Wichtig:
+		Vermeidung von Race Conditions:
+			Kein Zugriff von mehreren Prozessen gleichzeitig auf einen Speicher.
+			
+Posix:
+	Ist ein IEEE STandard zur einheitlichen Bereitstellung von Schnittstellen in Unixbasierten Systemen.
+	Dadurch ist es möglich dass ein Programm welches auf dem POSIX Standard basiert, höchstwahrscheinlich
+	auf einer vielzahl von Betriebssystemen laufen kann. Verwendet man Betriebssystem Spezifische API's,
+	wird man auf erhebliche Probleme stoßen, falls man die Anwendung auf ein anderes Betriebssystem portieren will.
+	
+	
+POSIX-Shell:
+	Die BASH ist nicht standardisiert. Sie wird durch die in ihr implementierten Funktionen
+	definiert. Die POSIX-Shell hingegen wird durch ihren Standard bestimmt, welcher es ermöglicht die POSIX-Shell
+	auf allen Unix Systemen zum Laufen zu bringen.
+	
+Shells:
+	Bourne (UNIX), POSIX, BASH, csh, zsh, cmd.exe, PowerShell etc
```

Comment:
Clarification needed:

1. What is the difference between Bourne (Unix) and BASH and POSIX?
2. What is `/bin/sh` and what is `#!/bin/sh` shebang?